pkg/util/validation/limits.go

```diff
@@ -192,6 +194,7 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&l.RequestBurstSize, RequestBurstSizeFlag, 0, "Per-tenant allowed push request burst size. 0 to disable.")
 	f.Float64Var(&l.IngestionRate, IngestionRateFlag, 10000, "Per-tenant ingestion rate limit in samples per second.")
 	f.IntVar(&l.IngestionBurstSize, IngestionBurstSizeFlag, 200000, "Per-tenant allowed ingestion burst size (in number of samples).")
+	f.IntVar(&l.IngestionBurstFactor, IngestionBurstFactorFlag, 0, "Per-tenant burst factor which is the maximum burst size allowed as a multiple of the per-tenant ingestion rate. If this is set it will override the `ingestion-burst-size` option.")
```

Comment:
I think it's simpler to keep this as a whole number, you'd always want this to be at least 1 so that your burst is larger than your ingest limit. You can make the case for having a range between 1 and 2 but I don't really see us tuning this number very much unlike the rate limit.